### PR TITLE
cherry-pick bug fixes in MKLDNN for v1.2.0

### PR DESCRIPTION
**Pull request by zheng-da**
## Description ##
This PR cherry-picks bug fixes for MKLDNN in the master branch to the v1.2.0 branch

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by zheng-da** [comment]
@anirudh2290

**Comment by anirudh2290** [comment]
Was #10591  mentioned in the dev ?

**Comment by pengzhao-intel** [comment]
@zheng-da please try if this issue (#11028) is gone in your branch.

**Comment by anirudh2290** [comment]
this pr is also making a change to the mkldnn submodule through https://github.com/apache/incubator-mxnet/pull/10578 . This worries me since the mkldnn submodule change has a change of 124 files and this makes it a really big change for the patch release. I am not comfortable including #10578. Do other PRs depend on the change ?

**Comment by zheng-da** [comment]
@anirudh2290 i have removed https://github.com/apache/incubator-mxnet/pull/10578
@pengzhao-intel it should work fine for https://github.com/apache/incubator-mxnet/issues/11028

**Comment by pengzhao-intel** [comment]
@anirudh2290 The update of MKL-DNN in #10578 fixed the depthConv issue.

**Comment by anirudh2290** [comment]
@pengzhao-intel I understand depthwise convolution is widely used but it is too big a change to be added to the patch release. What do others think @zheng-da @piiswrong ?

**Comment by pengzhao-intel** [comment]
@anirudh2290 Totally understand it will be a risk. 
The reason is MXNET used the specified CI of MKL-DNN. When we updated MKL-DNN to the bugfix CI, there are lots of other CIs so you see big changes. We have verified the performance of this CI, it works very well :)

**Comment by anirudh2290** [comment]
@pengzhao-intel There are still a lot of non CI-specific code changes. Since MKLDNN is marked as experimental for the release, I would err on the side of caution, add it to known issues for MKLDNN, and not add it to the release.

File: tests/cpp/operator/mkldnn.cc [hunk 1]
```diff
@@ -81,13 +81,6 @@ TEST(MKLDNN_UTIL_FUNC, AlignMem) {
 #endif
 }
 
-TEST(MKLDNN_UTIL_FUNC, MemFormat) {
-  // Check whether the number of format is correct.
-  CHECK_EQ(mkldnn_format_last, 56);
-  CHECK_EQ(mkldnn_nchw, 5);
-  CHECK_EQ(mkldnn_oihw, 12);
```
Comment (anirudh2290) [inline tests/cpp/operator/mkldnn.cc]: why are these unnecessary tests and how are they running on master ?

**Comment by zheng-da** [comment]
this test was from the PR
https://github.com/apache/incubator-mxnet/pull/10578/files.

**Comment by anirudh2290** [comment]
@zheng-da ok got it!

**Pull request merged**